common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -424,7 +430,12 @@ else if (codePoint == 0x03C2) {
    * @param target The target string to convert to lowercase.
    * @return The string converted to lowercase in a context-unaware manner.
    */
-  public static String lowerCaseCodePoints(final String target) {
+  public static UTF8String lowerCaseCodePoints(final UTF8String target) {
+    if (target.isFullAscii()) return target.toLowerCaseAscii();
+    return UTF8String.fromString(lowerCaseCodePoints(target.toString()));
+  }
+
+  private static String lowerCaseCodePoints(final String target) {
```

Comment:
Does it make sense to also make this accept and return `UTF8String` values?

Reply:
discussing in: https://github.com/apache/spark/pull/47014#discussion_r1652818328